src/codeu/chat/server/Server.java

```diff
@@ -57,19 +61,30 @@
   private final Uuid id;
   private final Secret secret;
 
-  private final Model model = new Model();
-  private final View view = new View(model);
   private final Controller controller;
 
   private final Relay relay;
   private Uuid lastSeen = Uuid.NULL;
+  private final Model model;
+  final View view;
 
-  public Server(final Uuid id, final Secret secret, final Relay relay) {
+  public Server(final Uuid id, final Secret secret, final Relay relay, final File fp) {
 
```

Comment:
File is a directory, so you can't wrote to it directly. You need to extend it to create a file underneath the directory, you can use this constructor to create a child file underneath the directory:
https://docs.oracle.com/javase/7/docs/api/java/io/File.html#File(java.io.File,%20java.lang.String)

This is the error I currently get when running the server:
INFO: Starting server...
java.io.FileNotFoundException: /tmp/data (Is a directory)
	at java.io.FileInputStream.open0(Native Method)
	at java.io.FileInputStream.open(FileInputStream.java:195)
	at java.io.FileInputStream.<init>(FileInputStream.java:138)
	at codeu.chat.server.Server.deserialize(Server.java:303)
	at codeu.chat.server.Server.<init>(Server.java:79)
	at codeu.chat.ServerMain.runServer(ServerMain.java:98)
	at codeu.chat.ServerMain.main(ServerMain.java:80)